src/Core/Liz.Core/License/EnrichPackageReferenceWithLicenseInformation.cs

```diff
@@ -0,0 +1,54 @@
+﻿using JetBrains.Annotations;
+using Liz.Core.Logging;
+using Liz.Core.PackageReferences;
+using System;
+using System.IO.Abstractions;
+using System.Threading.Tasks;
+
+namespace Liz.Core.License;
+
+internal sealed class EnrichPackageReferenceWithLicenseInformation : IEnrichPackageReferenceWithLicenseInformation
+{
+    private readonly IDownloadPackageReference _downloadPackageReference;
+    private readonly IGetLicenseInformationFromArtifact _getLicenseInformationFromArtifact;
+    private readonly ILogger _logger;
+
+    public EnrichPackageReferenceWithLicenseInformation(
+        [NotNull] IDownloadPackageReference downloadPackageReference,
+        [NotNull] IGetLicenseInformationFromArtifact getLicenseInformationFromArtifact,
+        [NotNull] ILogger logger)
+    {
+        _downloadPackageReference = downloadPackageReference ?? throw new ArgumentNullException(nameof(downloadPackageReference));
+        _getLicenseInformationFromArtifact = getLicenseInformationFromArtifact ?? throw new ArgumentNullException(nameof(getLicenseInformationFromArtifact));
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+    
+    public async Task GetFromPackageReferenceAsync(PackageReference packageReference)
```

Comment:
`GetFromPackageReferenceAsync` is not a great name for that method, change it to something like `EnrichAsync`